CONTRIBUTING.md

```diff
@@ -160,19 +220,70 @@ make html -b linkcheck
-PyVista out to PyPI and Conda-Forge for deployment. Please checkout the
-latest state of the `master` branch locally and:
-
-1. Run all tests as outlined in the [Testing Section](#testing) and ensure
-all are passing.
-
-2. Ensure all builds and tests on Azure are passing for the latest commits on
-the `master` branch
+#### Creating a New Pull Request
+
+Once you have tested your branch locally, push your branch to (pyvista
+GitHub)[https://github.com/pyvista/pyvista] and create a pull request
+while merging to master.  This will automatically run continious
+integration (CI) testing and verify your changes will work across
+several platforms.
+
+To ensure someone else reviews your code, at least one other member of
+the pyvista contributiors group must review and verify your code meets
+our community's standards.  Once approved, if you have write
+permission you may merge the branch.  If you don't have write
+permission, the reviewer or someone else with write permission will
+merge the branch and close out the PR.
+
+Since it may be necessary to merge your branch with the current
+release branch (see below), please do not close out your branch if it
+is a `fix/` branch.
+
+
+### Branching Model
+
+This project follows a modified
+[gitflow](https://datasift.github.io/gitflow/IntroducingGitFlow.html)
+branching model.  The purpose of this is to enable rapid development
+of features without sacrificing stability.  These are the main
+features:
+
+- The `master` branch is the main development branch.  All features,
+  patches, and other branches should be merged here.  While all PRs
+  should pass all applicable CI checks, this branch may be
+  functionally unstable as changes might have introduced unintended
+  side-effects or bugs that were not caught through unit testing.
+- There will be one or many `release/` branches based on minor
+  releases (for example `release/0.24`) which contain a stable version
+  of the code base that is also reflected on PyPi/.  Hotfixes from
+  `fix/` branches should be merged both to master and to these
+  branches.  When necessary to create a new patch release these
+  release branches will have their `__version__.py` updated and be
+  tagged with a patched semantic version (e.g. `0.24.1`).  This
+  triggers CI to push to PyPi, and allow us to rapidly push hotfixes
+  for past versions of `pyvista` without having to worry about
+  untested features.
+- When a minor release canidate is ready, a new `release` branch will
+  be created from `master` will be created with the next incremented
+  minor version (e.g. `release/0.25.0`), which will be throughly
+  tested.  When deemed stable, the release branch will be tagged with
+  the version (`0.25.0` in this case), and if necessary merged with
+  master if any changes were pushed to it.  Feature development then
+  continues on `master` and any hotfixes will now be merged with this
+  release.  Older release branches may be deleted when needed.
```

Comment:
I'd never delete a release branch. Sure, tags will stay to indicate past releases, but it's still a bit more obscure if they are on commits that aren't connected to any branches.

Reply:
Replaced last sentence with `Older release branches should not be deleted so they can be patched as needed.`